### PR TITLE
Prepare 0.24.0 release

### PR DESCRIPTION
**Pull request by djc**
- [x] Updated README (including changelog)
- [x] Dependencies are up to date
- [x] `cargo test --all-features` passes locally
- [x] Version in `Cargo.toml` is set to 0.24.0
- [x] `cargo publish --dry-run` works

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,6 +7,9 @@ and the [hyper HTTP library](https://github.com/hyperium/hyper).
 [![Documentation](https://docs.rs/hyper-rustls/badge.svg)](https://docs.rs/hyper-rustls/)
 
 # Release history
+- 0.24.0 (2023-04-01):
+  * Upgrade to rustls 0.21.0, tokio-rustls 0.24.0 and webpki-roots 0.23.0.
+  * Add `ConnectorBuilder::enable_all_versions()` helper.
```
Comment (cpu) [inline README.md:12]: Does the increase in MSRV merit a mention here? It went from 1.56.1 to 1.57 in https://github.com/rustls/hyper-rustls/commit/90d23a2bb75ec1193757e2d4f9d76b1308b8d844

**Review by cpu — APPROVED** [review]
Nothing in the [diff between main and 0.23.2](https://github.com/rustls/hyper-rustls/compare/v/0.23.2...main) stood out to me except possibly mentioning the MSRV  change.

Thanks @djc!

File: README.md (hunk 1 above)
Comment (djc) [reply README.md:12]: I don't think so? It will be meaningless to the vast majority of the audience.

**Comment by djc** [comment]
Published.

**Pull request merged**